src/lang/sclang.ts

```diff
@@ -1,28 +1,57 @@
+import { ChildProcess, spawn } from "child_process";
+import cuid from "cuid";
+import { EventEmitter } from "events";
+import fs from "fs";
+import yaml from "js-yaml";
+import _ from "lodash";
+import path from "path";
+import temp from "temp";
+import untildify from "untildify";
+
+import { SCError } from "../Errors";
+import { SclangResultType } from "../Types";
+import Logger from "../utils/logger";
+import { SclangCompileResult, SclangIO, State } from "./internals/sclang-io";
+
+interface SCLangOptions {
+  debug: boolean;
+  echo: boolean;
+  log?: Console;
+  // path to sclang executable
+  sclang: string;
+  // path to existing conf file
+  sclang_conf?: string;
+
+  stdin: boolean;
+  failIfSclangConfIsMissing: boolean;
+  conf: SCLangConf;
+}
+
+// import resolveOptions from "../utils/resolveOptions";
+export type SCLangArgs = Partial<SCLangOptions>;
+
+const defaults: SCLangOptions = {
+  debug: false,
+  echo: true,
+  // TODO resolve executable
+  sclang: "sclang",
+  failIfSclangConfIsMissing: false,
+  stdin: false,
+  conf: {
```

Comment:
Previously these were on the top level of the options.
- `includePaths`
- `excludePaths`

I'm not sure how important it is to preserve that. It would AFAICT the only breaking change